pkg/storage/sqlite/sqlite.go

```diff
@@ -0,0 +1,810 @@
+	})
+}
+
+// ReadUserTuple see [storage.RelationshipTupleReader].ReadUserTuple.
+func (m *SQLite) ReadUserTuple(ctx context.Context, store string, tupleKey *openfgav1.TupleKey, _ storage.ReadUserTupleOptions) (*openfgav1.Tuple, error) {
+	ctx, span := tracer.Start(ctx, "sqlite.ReadUserTuple")
+	defer span.End()
+
+	objectType, objectID := tupleUtils.SplitObject(tupleKey.GetObject())
+	userType := tupleUtils.GetUserTypeFromUser(tupleKey.GetUser())
+
+	var conditionName sql.NullString
+	var conditionContext []byte
+	var record storage.TupleRecord
+
+	err := m.stbl.
+		Select(
+			"object_type", "object_id", "relation", "_user",
+			"condition_name", "condition_context",
+		).
+		From("tuple").
+		Where(sq.Eq{
+			"store":       store,
+			"object_type": objectType,
+			"object_id":   objectID,
+			"relation":    tupleKey.GetRelation(),
+			"_user":       tupleKey.GetUser(),
+			"user_type":   userType,
+		}).
+		QueryRowContext(ctx).
+		Scan(
+			&record.ObjectType,
+			&record.ObjectID,
+			&record.Relation,
+			&record.User,
+			&conditionName,
+			&conditionContext,
+		)
+	if err != nil {
+		return nil, sqlcommon.HandleSQLError(err)
+	}
+
+	if conditionName.String != "" {
+		record.ConditionName = conditionName.String
+
+		if conditionContext != nil {
+			var conditionContextStruct structpb.Struct
+			if err := proto.Unmarshal(conditionContext, &conditionContextStruct); err != nil {
+				return nil, err
+			}
+			record.ConditionContext = &conditionContextStruct
+		}
+	}
+
+	return record.AsTuple(), nil
+}
+
+// ReadUsersetTuples see [storage.RelationshipTupleReader].ReadUsersetTuples.
+func (m *SQLite) ReadUsersetTuples(
+	ctx context.Context,
+	store string,
+	filter storage.ReadUsersetTuplesFilter,
+	_ storage.ReadUsersetTuplesOptions,
+) (storage.TupleIterator, error) {
+	ctx, span := tracer.Start(ctx, "sqlite.ReadUsersetTuples")
+	defer span.End()
+
+	sb := m.stbl.
+		Select(
+			"store", "object_type", "object_id", "relation", "_user",
+			"condition_name", "condition_context", "ulid", "inserted_at",
+		).
+		From("tuple").
+		Where(sq.Eq{"store": store}).
+		Where(sq.Eq{"user_type": tupleUtils.UserSet})
+
+	objectType, objectID := tupleUtils.SplitObject(filter.Object)
+	if objectType != "" {
+		sb = sb.Where(sq.Eq{"object_type": objectType})
+	}
+	if objectID != "" {
+		sb = sb.Where(sq.Eq{"object_id": objectID})
+	}
+	if filter.Relation != "" {
+		sb = sb.Where(sq.Eq{"relation": filter.Relation})
+	}
+	if len(filter.AllowedUserTypeRestrictions) > 0 {
+		orConditions := sq.Or{}
+		for _, userset := range filter.AllowedUserTypeRestrictions {
+			if _, ok := userset.GetRelationOrWildcard().(*openfgav1.RelationReference_Relation); ok {
+				orConditions = append(orConditions, sq.Like{"_user": userset.GetType() + ":%#" + userset.GetRelation()})
+			}
+			if _, ok := userset.GetRelationOrWildcard().(*openfgav1.RelationReference_Wildcard); ok {
+				orConditions = append(orConditions, sq.Eq{"_user": userset.GetType() + ":*"})
+			}
```

Comment:
There's an optimization possible in this code, that we didn't have time to bring to the other datastores, but you can bring it here if you like :) https://github.com/openfga/openfga/issues/1873

The gist would be to change:
- the `tuple` table definition
- `Write` method
- All the `Read*` methods involved in reading tuples
- This particular block would be 

```go
if _, ok := userset.GetRelationOrWildcard().(*openfgav1.RelationReference_Relation); ok {
	orConditions = append(orConditions, sq.Eq{"subject_relation": userset.GetRelation()})
	orConditions = append(orConditions, sq.Eq{"subject_type": userset.GetType()})
}
if _, ok := userset.GetRelationOrWildcard().(*openfgav1.RelationReference_Wildcard); ok {
	orConditions = append(orConditions, sq.Eq{"subject_id": "*"})
	orConditions = append(orConditions, sq.Eq{"subject_type": userset.GetType()})
}
```

Reply:
Will check it out 👍